packages/account/src/providers/transaction-request/upload-transaction-request.ts

```diff
@@ -0,0 +1,132 @@
+import { ZeroBytes32 } from '@fuel-ts/address/configs';
+import type { BytesLike } from '@fuel-ts/interfaces';
+import type { BN } from '@fuel-ts/math';
+import { type TransactionUpload, TransactionType } from '@fuel-ts/transactions';
+import { arrayify, hexlify } from '@fuel-ts/utils';
+
+import type { GqlGasCosts } from '../__generated__/operations';
+import { calculateMetadataGasForTxUpload } from '../utils';
+
+import { hashTransaction } from './hash-transaction';
+import { type BaseTransactionRequestLike, BaseTransactionRequest } from './transaction-request';
+
+/**
+ * @hidden
+ */
+export interface UploadSubsectionRequest {
+  /** The root of the Merkle tree is created over the bytecode. */
+  root: BytesLike;
+
+  /** The index of the subsection of the bytecode. */
+  subsectionIndex: number;
+
+  /** The total number of subsections on which bytecode was divided. */
+  subsectionsNumber: number;
+
+  /** The proof set helps to verify the connection of the subsection to the `root`. */
+  proofSet: BytesLike[];
+
+  /** The subsection bytecode. */
+  subsection: BytesLike;
+}
+
+/**
+ * @hidden
+ */
+export interface UploadTransactionRequestLike extends BaseTransactionRequestLike {
+  /** The witness index of the subsection of the bytecode. */
+  witnessIndex?: number;
+
+  /** The subsection data. */
+  subsection?: Omit<UploadSubsectionRequest, 'subsection'>;
+}
+
+export class UploadTransactionRequest extends BaseTransactionRequest {
+  static from(obj: UploadTransactionRequestLike): UploadTransactionRequest {
+    if (obj instanceof UploadTransactionRequest) {
+      return obj;
+    }
+    return new UploadTransactionRequest(obj);
```

Comment:
Please return as deep clone as in https://github.com/FuelLabs/fuels-ts/pull/3097.